### PR TITLE
Data filtering

### PR DESCRIPTION
**Pull request by Anakin100100**

### CODE REVIEW

File: app/controllers/institutions_controller.rb
Comment (wojtodzio) [inline app/controllers/institutions_controller.rb:65]: This name can be confusing; something like `public_school` would probably be a better fit for a boolean.

**Review by wojtodzio — CHANGES_REQUESTED** [review]

File: Dockerfile
Comment (Anakin100100) [reply Dockerfile]: it was left from debugging

File: db/schema.rb
Comment (Anakin100100) [reply db/schema.rb:85]: These names are specific to the polish law and I don't know an official translation for them. I don't think that there is a one to one translation for them that can be universally understood. For example a word "county" could be used to replace any of the above and it wouldn't be a translation mistake. Leaving them in polish allows us to see exctly what they mean. If any foreigners (highly unlikely) encounter them they would have to familiarise themselves at least somewhat with how the polish administrative division regardless of the name in the database. Can you think of a better naming convention?

Comment (Anakin100100) [reply db/schema.rb:85]: I think that it allows for a more granular search for a user. In some cases only the miejscowosc field has any meaning and we are searching on all three of those. If a user were to search for school only inside a specific miejscowosc that is not a powiat or a gmina that search would return nothing even though schools exist in that area.

File: docs/api.md
Comment (Anakin100100) [reply docs/api.md:22]: Implementing rswag would require rewriting all the integration tests first and later adding that. There 10 lines of comment make it clear how to use the api for Michał.  Shouldn't we create another pull request focused on rswag in order not to create one massive coupled pull request here?

Comment (wojtodzio) [reply docs/api.md]: Makes sense 👍

File: db/schema.rb
Comment (wojtodzio) [reply db/schema.rb:85]: I'm not sure what's the best solution here, but having both "city" and "miejscowosc" in the same table is really confusing for me. If we need to have both, then we should name them in a way where it's clear what's the difference between them. It seems like "city" is taken from the "adresDoKorespondecjiMiejscowosc" field, while "miejscowosc" uses the "miejscowosc" field, so maybe we should use a similar convention, e.g., "postalAddressCity" and "city"?

Comment (wojtodzio) [reply db/schema.rb:85]: Yeah, good point. According to Wikipedia, "gmina" is most similar to "municipality" (https://en.wikipedia.org/wiki/Gmina), while "powiat" to "county" (https://en.wikipedia.org/wiki/Powiat), so maybe that's a way to go?
Another way of thinking about it could be that both "gmina" and "powiat" are both examples of specific levels of what's called administrative divisions (see list at https://en.wikipedia.org/wiki/List_of_administrative_divisions_by_country), so maybe we could have something like `administrative_division_level_1`, `administrative_division_level_2` etc.? That way, this part of our model would be country agnostic, which would be very cool, as that would make future app's globalization easier and can potentially attract contributors from other countries ;)

Comment (Anakin100100) [reply db/schema.rb:79]: I forgot that we had the "city" column there. I'm going to delete it.

Comment (Anakin100100) [reply db/schema.rb:78]: We can start thinking about making it country agnostic when we specify which country we would like to expand to. Let's focus for now on making it in Poland.

**Comment by Anakin100100** [comment]
@micorix please see if the latest version of dockerfile there has solved your local build issue.

File: app/controllers/institutions_controller.rb
Comment (wojtodzio) [inline app/controllers/institutions_controller.rb:37]: You can render in the before_action instead of relying on a temp. instance variable
```suggestion
    render status: :bad_request, json: { message: 'page must be greater than 0' } if @page < 1
```

File: app/jobs/create_institution_records_job.rb
Comment (wojtodzio) [inline app/jobs/create_institution_records_job.rb:35]: So what's the final verdict when it comes to those names? I previously suggested two solutions: either using English names to stay consistent with the rest of the system or introducing an abstraction in the form of general administrative divisions.

File: db/schema.rb
Comment (wojtodzio) [inline db/schema.rb:89]: That's strange. I wonder where are those changes coming from?

**Review by wojtodzio — CHANGES_REQUESTED** [review]
Nice, almost there

File: db/schema.rb
Comment (Anakin100100) [reply db/schema.rb:89]: This must have come from some new cop inside rubocop. I didn't introduce those changes manually and I didn't use any external commands that might have modified the schema this way. This seems to affect all datetime fields except created_at and updated_at in all tables.

Comment (Anakin100100) [reply db/schema.rb:89]: The same cop has removed specified precision from all created_at and updated_at fields in all tables.

File: app/jobs/create_institution_records_job.rb
Comment (Anakin100100) [reply app/jobs/create_institution_records_job.rb]: let's translate them to English

File: db/schema.rb
Comment (wojtodzio) [reply db/schema.rb:89]: We an disable rubocop for schema

**Review by wojtodzio — APPROVED** [review]
Nice job! 🚀

File: db/schema.rb
Comment (Anakin100100) [reply db/schema.rb:89]: Now that you mention that I noticed that we already did. does 
excluding 
    - 'db/**/*'
not exclude files in db directory but only files in subdirectiories in the db directory? Should we specifically exclude the schema file?

File: .ci/docker-compose.test.yml
Comment (micorix) [inline .ci/docker-compose.test.yml:5]: Whole image name is not tag per se. It could be confusing.

File: .ci/Dockerfile.postgres
Comment (micorix) [inline .ci/Dockerfile.postgres:1]: Why? We can just add `postgres:13` to docker-compose file.

File: .ci/docker-compose.test.yml
Comment (micorix) [inline .ci/docker-compose.test.yml:10]: The approach we took in po8klasie/po8klasie and po8klasie/po8klasie-api repo is that we had two entrypoints files (one prod-ready, one for linters and testing) and we would overwrite command here to choose entrypoint for testing.

Why this approach might be better?
IMO it would better if linting work within the docker, cause 1. it'll have the exact version as files in repo, 2. starting another action took pricey minutes, 3. it would cancel this job (so cancel tests) if linting fails.

Comment (micorix) [reply .ci/docker-compose.test.yml:10]: I mean expressing commands directly in yaml here is fine. It just mostly about linting within container

File: .ci/docker-compose.test.yml
Comment (micorix) [inline .ci/docker-compose.test.yml:14]: As stated before

File: .ci/docker-compose.test.yml
Comment (micorix) [inline .ci/docker-compose.test.yml:16]: It doesn't matter IMO here, bu just a reminder we can always use networks instead of exposing ports

File: .github/workflows/lint.yml
Comment (micorix) [inline .github/workflows/lint.yml:13]: Pls see comments in docker-compose file

File: .github/workflows/test.yml
Comment (micorix) [inline .github/workflows/test.yml:3]: Why not just on PRs? As we have protected master branch, we don't need to use minutes on testing code after merging it.
Workflow for pushing would also trigger prod deploy, so it'll better to keep them separate.

File: .github/workflows/test.yml
Comment (micorix) [inline .github/workflows/test.yml:54]: We don't have to build it. See comment in db dockerfile

Comment (micorix) [reply .github/workflows/test.yml:54]: Also we don't need to cache it either (if im not wrong idk)

**Review by micorix — CHANGES_REQUESTED** [review]
Hi. Thanks! Just a few comments

File: docs/api.md
Comment (micorix) [inline docs/api.md:8]: Shouldn't that be changed also in swagger config file? idk just guessing

Comment (Anakin100100) [reply docs/api.md:8]: We should but developong swagger is kinda on hold because the single markdown file does the job fine.

**Review by micorix — APPROVED** [review]
docker part lgtm

**Pull request merged**
